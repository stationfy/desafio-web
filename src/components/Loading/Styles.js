import styled, { keyframes } from "styled-components";
import { FaSpinner } from "react-icons/fa";

export const Container = styled.div`
  min-height: 70vh;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const SpinToWin = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

export const Spinner = styled(FaSpinner)`
  animation: ${SpinToWin} 2s infinite linear;
`;
