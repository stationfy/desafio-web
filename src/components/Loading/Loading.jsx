import React from "react";

import LoadingIcon from "../../assets/loading.svg";

import { Container, Spinner } from "./Styles";

const Loading = () => {
  return (
    <Container>
      <Spinner src={LoadingIcon} alt="Carregando" />
    </Container>
  );
};

export default Loading;
