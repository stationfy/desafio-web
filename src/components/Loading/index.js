import React from "react";

import { Container, Spinner } from "./Styles";

const Loading = () => {
  return (
    <Container>
      <Spinner size={90} />
    </Container>
  );
};

export default Loading;
