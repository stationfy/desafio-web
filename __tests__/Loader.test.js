import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../src/components/loader/Loading';

describe('Testing Loading Component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <Loading />
        );

        expect(wrapper).toMatchSnapshot();
    });
});