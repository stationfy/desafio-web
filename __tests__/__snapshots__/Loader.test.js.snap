// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Loading Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <FaSpinner
        fontSize={25}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "fontSize": 25,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <FaSpinner
          fontSize={25}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fontSize": 25,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
